search/client.go

```diff
@@ -59,15 +63,12 @@ func (c *client) Query(q Query) (*Response, error) {
 	}
 	buf := bytes.NewReader(marshalledQuery)
 
-	req, err := http.NewRequest("POST", c.endpoint, buf)
-	if err != nil {
-		return nil, err
+	headers := map[string]string{
```

Comment:
Please do not create a new map when you already have one created when making a new Request.